### DISCUSSION TITLE
Should DgsDataFetchingEnvironment or DataFetchingEnvironment be used as query/mutation argument?

### DISCUSSION THREAD

**Discussion by carmas-RatedPower** [Netflix/dgs-framework#624] [Q&A] [answered]
This might be a kind of dumb question but, while reading the docs, I've noticed that in some query and mutation examples, sometimes a DgsDataFetchingEnvironment is passed as method argument and some others pass a DataFetchingEnvironment. Just wanted to know which is the scenario or advantage to pass the implementation instead of the interface.

Links to the docs to better explain my question:

* Passing DgsDataFetchingEnvironment:
    * https://netflix.github.io/dgs/datafetching/#using-dgsrequestdata
```
@DgsData(parentType = "Query", field = "serverName")
public String serverName(DgsDataFetchingEnvironment dfe) {
     DgsRequestData requestData =  DgsContext.getRequestData(dfe);
     return ((ServletWebRequest)requestData.getWebRequest()).getRequest().getServerName();
}
```

* Passing DataFetchingEnvironment:
    * https://netflix.github.io/dgs/mutations/
```
@DgsComponent
public class RatingMutation {
    @DgsData(parentType = "Mutation", field = "addRating")
    public Rating addRating(DataFetchingEnvironment dataFetchingEnvironment) {
        int stars = dataFetchingEnvironment.getArgument("stars");

        if(stars < 1) {
            throw new IllegalArgumentException("Stars must be 1-5");
        }

        String title = dataFetchingEnvironment.getArgument("title");
        System.out.println("Rated " + title + " with " + stars + " stars") ;

        return new Rating(stars);
    }
}
```

**Comment by carmas-RatedPower** [comment]
Any suggestions?

**Comment by paulbakker** [comment, marked as answer]
The `DataFetchingEnvironment` comes from the `graphql-java` Library. The DGS framework extends it to add some convenience methods.
If you don’t need the convenience methods, it really doesn’t matter which one you use.
Whichever type you use, it will receive the same instance at runtime.

I find myself defaulting to `DgsDataFetchingEnvironment` most of the time, but haven’t been very consistent about it.
